src/context/quizContext.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { createContext, useContext, useReducer } from 'react';
+import { quizdatabase } from '../database/database';
+import { State } from './quizContext.type';
+import { quizReducer } from './quizReducer';
+
+export const initialStates: State = {
+	quiz: quizdatabase,
+	currentQuestionNumber: -1,
+	score: 0,
+	result: {
+		quizId: '',
+		resultArray: [],
+	},
+	currentQuiz: null,
+};
+const AppContext = createContext<{
+	quizState: State;
+	quizDispatch: React.Dispatch<any>;
+}>({
+	quizState: initialStates,
+	quizDispatch: () => null,
+});
```

Comment:
You could use type assertion to avoid the typeScript errors for initail state.
```javascript
const AppContext = createContext<ContextType>({} as ContextType)
```

Reply:
Done, Thanks for this